lib/improver/percentile.py

```diff
@@ -0,0 +1,149 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing percentiling classes."""
+
+
+import iris
+from iris.exceptions import CoordinateNotFoundError
+from iris import FUTURE
+
+FUTURE.netcdf_promote = True
+
+
+class PercentileConverter(object):
+
+    """Plugin for converting from a set of values to a PDF.
+
+    Generate percentiles together with min, max, mean, stdev.
+
+    """
+
+    # Default percentile boundaries to calculate at.
+    DEFAULT_PERCENTILES = [0, 5, 10, 20, 25, 30, 40, 50,
+                           60, 70, 75, 80, 90, 95, 100]
+
+    def __init__(self, collapse_coord, percentiles=None):
+        """Create a PDF plugin with a given source plugin.
+
+        Parameters
+        ----------
+        collapse_coord : str or iris.coord.DimCoord (or list of either)
+            The name or coordinate definition of the coordinate(s) to collapse
+            over.
+
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES.
+
+        """
+        if not isinstance(collapse_coord, list):
+            collapse_coord = [collapse_coord]
+        for coord_item in collapse_coord:
+            _type_test(coord_item)
+
+        if percentiles is not None:
+            self.percentiles = [int(value) for value in percentiles]
+        else:
+            self.percentiles = self.DEFAULT_PERCENTILES
+
+        # Collapsing multiple coordinates results in a new percentile
+        # coordinate, its name suffixed by the original coordinate names. Such
+        # a collapse is cummutative (i.e. coordinate order doesn't matter).
+        # However the coordinates are sorted such that the resulting
+        # percentile coordinate has a consistent name regardless of the order
+        # in which the user provides the original coordinate names.
+        self.collapse_coord = sorted(collapse_coord, _sort_coord)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<PercentileConverter: collapse_coord={}, percentiles={}'
+                .format(self.collapse_coord, self.percentiles))
+        return desc
+
+    def process(self, cube):
+        """Create a cube containing the percentiles as a new dimension.
+
+        What's generated is:
+            * 13 percentiles - (5%, 10%, 20%, 25%, 30%, 40%, 50%, 60%,
```

Comment:
I notice that this differs from the default percentiles, as there's no 0 and 100 percentile. Is this correct?